samples/cellular/nrf_cloud_rest_fota/src/main.c

```diff
@@ -771,68 +417,23 @@ int main(void)
 	while (1) {
 
 		/* Generate a JWT for each REST call */
-		err = generate_jwt();
-		if (err) {
-			error_reboot();
-		}
-
-		/* If a FOTA job is in progress, handle it first */
-		if (validate_in_progress_job()) {
-			err = update_job_status();
-			if (err) {
-				error_reboot();
-			}
-
-			wait_after_job_update();
-			continue;
-		}
-
-		/* Check for a new FOTA job */
-		err = check_for_job();
-		if (err < 0) {
-			error_reboot();
-		} else if (err > 0) {
-			/* No job. Wait for the configured duration or a button press */
-			cleanup();
-			(void)nrf_cloud_rest_disconnect(&rest_ctx);
-
-			LOG_INF("Retrying in %d minute(s) or when button %d is pressed",
-				CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN,
-				CONFIG_REST_FOTA_BUTTON_EVT_NUM);
-
-			(void)k_sem_take(&button_press_sem,
-				K_MINUTES(CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN));
-			continue;
-		}
-
-		/* Start the FOTA download process and wait for completion (or timeout) */
-		err = start_download();
+		err = generate_jwt(&rest_ctx);
 		if (err) {
-			LOG_ERR("Failed to start FOTA download");
-			error_reboot();
+			error_reboot(FOTA_REBOOT_SYS_ERROR);
 		}
 
-		err = wait_for_download();
-		if (err == -ETIMEDOUT) {
-			LOG_ERR("Timeout; FOTA download took longer than %d minutes",
-				CONFIG_REST_FOTA_DL_TIMEOUT_MIN);
-			fota_status = NRF_CLOUD_FOTA_TIMED_OUT;
-			fota_status_details = FOTA_STATUS_DETAILS_TIMEOUT;
+		err = nrf_cloud_fota_support_process(&fota_ctx);
```

Comment:
since this is a sample, it might be good to add comments here describing what all this function does.
for example, it can block for awhile while the download occurs.